packages/synapse-react-client/src/components/entity/metadata/EntityModal.tsx

```diff
@@ -105,15 +126,11 @@ export function EntityModal(props: EntityModalProps) {
     <Button
       variant={'outlined'}
       onClick={() => {
-        if (hasClickedCancel) {
-          setIsInEditMode(false)
-        }
-        setHasClickedCancel(!hasClickedCancel)
+        setOnCancelEdit(() => () => setIsInEditMode(false))
+        setShowEditCancellationWarning(true)
```

Comment:
If this button is clicked, we want to disable edit mode, but not close the modal